BookingProject/BookingWebAPI/BookingWebApi.Application/Booking/BookingService.cs

```diff
@@ -0,0 +1,98 @@
+﻿using AutoMapper;
+using BookingWebApi.Application.Apartament;
+using Contracts.Clients;
+using Contracts.DTOs.Audit;
+using Contracts.DTOs.Payment;
+using Microsoft.Extensions.Logging;
+using Response;
+using Saga;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using BookingEntity = BookingWebApi.Domain.Entities.Booking;
+
+namespace BookingWebApi.Application.Booking
+{
+    public interface IBookingService
+    {
+        Task<Result<BookingDto>> CreateBooking(string userId, BookingDto bookingDto);
+    }
+
+    public class BookingService(
+        IBookingRepository _bookingRepository,
+        IApartamentRepository _apartamentRepository,
+        IPaymentClient _paymentClient,
+        IAuditClient _auditClient,
+        IMapper _mapper,
+        ILogger<BookingService> _logger
+    ): IBookingService
+    {
+        public async Task<Result<BookingDto>> CreateBooking(string userId, BookingDto bookingDto)
+        {
+            var booking = _mapper.Map<BookingEntity>(bookingDto);
+            booking.UserId = userId;
+
+            var res = (await _apartamentRepository.CalculateTotalPriceWithCurrency(
+                    bookingDto.ApartamentId, bookingDto.StartDate, bookingDto.EndDate));
+
+            if (!res.IsSuccess) return Result<BookingDto>.Failure(res.ErrorMessage);
+
+            booking.TotalAmount = res.Value.TotalPrice;
+            booking.CurrencyName = res.Value.CurrencyName;
+            booking.Status = "Creating";
```

Comment:
Why would we always allocate memory for strings? Better use enums and they are better to store in DB as status